### PR TITLE
Refactor WoolMatchModule into new text library

### LINKED ISSUE #508: Broken translation message for uncraftable wools
Wools with the uncraftable attribute display the following message when a player attempts to craft a wool 
image

### LINKED ISSUE #509: Issues on scoreboard rendering compact wools
image

In this map, the scoreboard should be using the "highly compact" mode, where it removes lines between teams, to be able to fit the last team, also, there seems to be a single space between each wool in the same row, this should be 2 spaces because it's only displaying 3 wools so it can fit

### PR DESCRIPTION
**Pull request by KingOfSquares**
* Also prepares Party, Goal and Team for refactor.

Fixes #508 

I hesitated refactoring anymore since #20 is assigned to Electroid
is it something to help with? Or do i leave it alone

Signed-off-by: KingOfSquares <simonmorland@gmail.com>

### CODE REVIEW

File: core/src/main/java/tc/oc/pgm/wool/WoolMatchModule.java [hunk 1]
```diff
@@ -183,14 +180,13 @@ public void placementCheck(final BlockTransformEvent event) {
 
     ParticipantState player = ParticipantBlockTransformEvent.getPlayerState(event);
     if (player != null) { // wool can only be placed by a player
-      Component woolName = wool.getComponentName();
+      net.kyori.text.Component woolName = wool.getTextName();
       if (!isValidWool(wool.getDyeColor(), event.getNewState())) {
-        player.sendWarning(new PersonalizedTranslatable("wool.wrongWool", woolName), true);
+        player.sendWarning(TranslatableComponent.of("wool.wrongWool").args(woolName));
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/wool/WoolMatchModule.java:185]: `TranslatableComponent.of("wool.wrongWool", woolName)` also works

File: core/src/main/java/tc/oc/pgm/wool/WoolMatchModule.java [hunk 2]
```diff
@@ -219,12 +215,8 @@ public void handleWoolCrafting(PrepareItemCraftEvent event) {
         for (MonumentWool wool : this.wools.values()) {
           if (wool.getDefinition().isObjectiveWool(result)) {
             if (!wool.getDefinition().isCraftable()) {
-              playerHolder.sendMessage(
-                  ChatColor.RED
-                      + TextTranslations.translate(
-                          "wool.craftingDisabled",
-                          playerHolder.getBukkit(),
-                          wool.getComponentName()));
+              playerHolder.sendWarning(
```
Comment (Electroid) [inline core/src/main/java/tc/oc/pgm/wool/WoolMatchModule.java:218]: Same here.

**Comment by KingOfSquares** [comment]
I updated this just to fix the bug mentioned, i started on some other stuff but its not done yet.

**Comment by Electroid** [comment]
See #519

**Pull request closed without merging**